Issues found when building with MSVC + /permissive- - 2

It is similar to the second issue in https://github.com/Microsoft/ChakraCore/issues/2862.

ch.cpp(877): error C2362: initialization of 'initRpcServer' is skipped by 'goto cleanup'

Revision https://github.com/Microsoft/ChakraCore/commit/e8f31729a0e8f04a68269698680657698e918053#diff-9aa417ac81c06da693354e50d1edd449 introduces 'goto', but this skips the local variable 'initRpcServer' and is ill-formed.

Either use RAII or put the variable into a separate scope.

@xiaoyinl

--- a/bin/ch/ch.cpp
+++ b/bin/ch/ch.cpp
@@ -846,6 +846,7 @@ int _cdecl RunJITServer(int argc, __in_ecount(argc) LPWSTR argv[])
     HINSTANCE chakraLibrary = nullptr;
     bool success = ChakraRTInterface::LoadChakraDll(&argInfo, &chakraLibrary);
     int status = 0;
+    JsInitializeJITServerPtr initRpcServer = nullptr;
 
     if (!success)
     {
@@ -860,7 +861,7 @@ int _cdecl RunJITServer(int argc, __in_ecount(argc) LPWSTR argv[])
         goto cleanup;
     }
 
-    JsInitializeJITServerPtr initRpcServer = (JsInitializeJITServerPtr)GetProcAddress(chakraLibrary, "JsInitializeJITServer");
+    initRpcServer = (JsInitializeJITServerPtr)GetProcAddress(chakraLibrary, "JsInitializeJITServer");
     status = initRpcServer(&connectionUuid, nullptr, nullptr);
     if (FAILED(status))
     {

